plugins/repository-azure/src/test/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceMock.java

```diff
@@ -94,8 +94,15 @@ public OutputStream getOutputStream(String account, LocationMode mode, String co
     public Map<String, BlobMetaData> listBlobsByPrefix(String account, LocationMode mode, String container, String keyPath, String prefix) {
         MapBuilder<String, BlobMetaData> blobsBuilder = MapBuilder.newMapBuilder();
         for (String blobName : blobs.keySet()) {
-            if (startsWithIgnoreCase(blobName, prefix)) {
-                blobsBuilder.put(blobName, new PlainBlobMetaData(blobName, blobs.get(blobName).size()));
+            final String checkBlob;
+            if (keyPath != null) {
+                // strip off key path from the beginning of the blob name
+                checkBlob = blobName.replace(keyPath, "");
```

Comment:
Because the blobNames are named with the directory path preceding the blob name.  So a blob named `foobar` in the repository will actually have a blob name of `/some/azure/path/foo`.  This causes issues because if I wanted to list all blob names in the mock repository that start with `foo`, none would match because they all start with the key path (i.e. `/some/azure/path`).  This basically makes it easier to work with the blob names in this mock.  Its worth revisiting whether we need the key path at all to be stored in the blob names for this mock repository, but I think that's a separate issue that needs a separate PR.